docs/DeveloperGuide.md

```diff
@@ -149,103 +149,104 @@ Classes used by multiple components are in the `seedu.storemando.commons` packag
+    Then, a `ClearCommandParser` object is created to further parse the command.
+4. `StoreMandoParser` then calls on the `parse` method of `ClearCommandParser` to parse the arguments provided.
+5. `ClearCommandParser` checks if the argument provided is an empty string. If so, the constructor of `ClearCommand` 
+    without any parameters is called. Else, a `LocationContainsPredicate` object will be created with the arguments 
+    as parameter.`CommandParseException` will be thrown if this is not possible.
+6. `ClearCommand` object will be returned to `ClearCommandParser` which then returns it to `LogicManager`. `LogicManager`
+    then calls the `execute` method of `ClearCommand` with a `Model` object, model, as parameter.
+7. `ClearCommand` calls on the `clearLocation` method of model with the `ClearCommand`'s attribute predicate as parameter. 
+    Subsequently, it calls on model's `updateFilteredItemList`.
+8. Finally, a `CommandResult` object is created and is returned to `LogicManager`.   
 
-</div>
 
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the inventory, such as `list`,
-will usually not call `Model#commitStoreMando()`, `Model#undoStoreMando()` or `Model#redoStoreMando()`. Thus,
-the `storeMandoStateList` remains unchanged.
+The following sequence diagram shows how clear feature works:
 
-![UndoRedoState4](images/UndoRedoState4.png)
+![ClearSequenceDiagram](images/ClearSequenceDiagram.png)
 
-Step 6. The user executes `clear`, which calls `Model#commitStoreMando()`. Since the `currentStatePointer` is not
-pointing at the end of the `storeMandoStateList`, all inventory states after the `currentStatePointer` will be purged.
-Reason: It no longer makes sense to redo the `add n/David …​` command. This is the behavior that most modern desktop
-applications follow.
+The following sequence diagram shows how a clear by location feature works:
 
-![UndoRedoState5](images/UndoRedoState5.png)
+![ClearLocationSequenceDiagram](images/ClearLocationSequenceDiagram.png)
 
 The following activity diagram summarizes what happens when a user executes a new command:
 
-![CommitActivityDiagram](images/CommitActivityDiagram.png)
+![ClearActivityDiagram](images/ClearLocationActivityDiagram.png)
 
 #### Design consideration:
 
-##### Aspect: How undo & redo executes
-
-* **Alternative 1 (current choice):** Saves the entire inventory.
-    * Pros: Easy to implement.
-    * Cons: May have performance issues in terms of memory usage.
-
-* **Alternative 2:** Individual command knows how to undo/redo by itself.
-    * Pros: Will use less memory (e.g. for `delete`, just save the item being deleted).
-    * Cons: We must ensure that the implementation of each individual command are correct.
-
-_{more aspects and alternatives to be added}_
+##### Aspect: How `clear` executes
 
-### \[Proposed\] Data archiving
+**Alternative 1 (current choice)** : 
+* Pros: 
```

Comment:
Is this incomplete? If not, I think it is best to omit the section altogether if there is nothing to add.